### PR TITLE
[Conan]Update Conan, CMake and Ninja;Fix Conan builds

### PR DESCRIPTION
**Pull request by vmalia**
# Description

*Update Ninja to 1.10.0 to support FORTRAN builds (Fixes Netlib LAPACK builds via Conan)
*Update CMake to 3.15 to support Ninja 1.10.0
*Update Conan to 1.24 to support OpenSSL package(new dependency for CMake package)
*Disable Conan compiler check to avoid failure when using DPC++ compiler

### CODE REVIEW

File: conan/profiles/inteldpcpp_lnx [hunk 1]
```diff
@@ -30,8 +30,8 @@ compiler.libcxx=libstdc++11
 build_type=Release
 
 [build_requires]
-ninja/1.9.0
-cmake/[>=3.13]
+cmake/[>=3.15]
+ninja/1.10.0
```
Comment (mkrainiuk) [inline conan/profiles/inteldpcpp_lnx:34]: We need to update supported versions in README as well

File: conanfile.py [hunk 2]
```diff
@@ -133,7 +133,9 @@ def build(self):
             "BUILD_DOC"                : self.options.build_doc,
 
             # Paramaters
-            "MKL_ROOT"                 : f"/opt/intel/inteloneapi/mkl/{self.oneapi_version}",
+            # Conan does not officially support the DPC++ compiler, hence disable the compiler_id check
+            "CONAN_DISABLE_CHECK_COMPILER" : True,
+            "MKL_ROOT"                 : f"/opt/intel/inteloneapi/mkl/latest",
```
Comment (mkrainiuk) [inline conanfile.py:138]: I see the same changes for MKL_ROOT in PR #29. Do you plan to close it and extend this one with other changes or keep them separated?

Comment (vmalia) [reply conanfile.py]: I will keep the PRs separate. Reverted that change in this PR.

File: conan/profiles/inteldpcpp_lnx (hunk 1 above)
Comment (vmalia) [reply conan/profiles/inteldpcpp_lnx:34]: Done

**Pull request merged**